frontend/src/pages/retailer/ModalAddWarehouseRetailer.js

```diff
@@ -0,0 +1,228 @@
+import React, {useState} from 'react';
+import Modal from "react-bootstrap/Modal";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import ErrorMessage from "../../messages/errorMessage";
+import {AsyncTypeahead} from "react-bootstrap-typeahead";
+import validateWarehouse from "../../validation/WarehouseValidationRules";
+
+function ModalAddWarehouseRetailer(props) {
+    const currentCustomerId = localStorage.getItem("currentCustomerId") != null ? localStorage.getItem("currentCustomerId") : 0;
+
+    const ref = React.createRef();
+    const [dropdownMenuName, setDropdownMenuName] = useState("select type");
+    const [warehouseDto, setWarehouseDto] = useState({
+        id: '',
+        customerId: currentCustomerId,
+        identifier: '',
+        type: 'RETAILER',
+        addressDto: {
+            state: {}
+        },
+        totalCapacity: ''
+    });
+
+    const [errors, setErrors] = useState({
+        validationErrors: [],
+        serverErrors: ''
+    });
+
+    const [options, setOptions] = useState([]);
+
+    const filterBy = () => true;
+
+    const handleSearch = (query) => {
+        fetch(`/customers/${currentCustomerId}/states?state=${query}`)
+            .then(resp => resp.json())
+            .then(res => {
+                setOptions(res);
+            });
+    };
+
+    const onChangeState = (e) => {
+        setErrors({
+            setErrors: '',
+            validationErrors: []
+        });
+        e.length > 0 ?
```

Comment:
we may put this check (e.length will be enough, no need to add > 0) to this part state: {id: e[0].id, state: e[0].stateZone}.
Will be something like this 
            setWarehouseDto(preState => ({
                ...preState,
                addressDto: {...preState.addressDto, state: e.length ? {id: e[0].id, state: e[0].stateZone} : {id: '', state: ''}}
            }))